images.CI/linux-and-win/azure-pipelines/image-generation.yml

```diff
@@ -71,28 +71,71 @@ jobs:
+
+        $TempResourceGroupName = "packer-temp-$ManagedImageName"
+        Write-Host "##vso[task.setvariable variable=TempResourceGroupName;]$TempResourceGroupName"
+
   - task: PowerShell@2
     displayName: 'Build VM'
     inputs:
       targetType: filePath
       filePath: ./images.CI/linux-and-win/build-image.ps1
-      arguments: -ResourcesNamePrefix $(Build.BuildId) `
-                        -ClientId $(CLIENT_ID) `
-                        -ClientSecret $(CLIENT_SECRET) `
-                        -TemplatePath $(TemplatePath) `
-                        -ResourceGroup $(AZURE_RESOURCE_GROUP) `
-                        -StorageAccount $(AZURE_STORAGE_ACCOUNT) `
-                        -SubscriptionId $(AZURE_SUBSCRIPTION) `
-                        -TenantId $(AZURE_TENANT) `
-                        -Location $(AZURE_LOCATION) `
-                        -VirtualNetworkName $(BUILD_AGENT_VNET_NAME) `
-                        -VirtualNetworkRG $(BUILD_AGENT_VNET_RESOURCE_GROUP) `
-                        -VirtualNetworkSubnet $(BUILD_AGENT_SUBNET_NAME)
+      arguments: -ClientId $(CLIENT_ID) `
+                 -ClientSecret $(CLIENT_SECRET) `
+                 -TemplatePath $(TemplatePath) `
+                 -ImageName "$(ManagedImageName)" `
+                 -ImageResourceGroupName $(AZURE_RESOURCE_GROUP) `
+                 -TempResourceGroupName "$(TempResourceGroupName)" `
+                 -SubscriptionId $(AZURE_SUBSCRIPTION) `
+                 -TenantId $(AZURE_TENANT) `
+                 -Location $(AZURE_LOCATION) `
+                 -VirtualNetworkName $(BUILD_AGENT_VNET_NAME) `
+                 -VirtualNetworkRG $(BUILD_AGENT_VNET_RESOURCE_GROUP) `
+                 -VirtualNetworkSubnet $(BUILD_AGENT_SUBNET_NAME)
 
     env:
       PACKER_LOG: 1
       PACKER_LOG_PATH: "$(Agent.TempDirectory)/packer-log.txt"
-  
+
+  - task: PowerShell@2
+    displayName: 'Convert managed image to VHD'
+    inputs:
+      targetType: filePath
+      filePath: ./images.CI/linux-and-win/convert-to-vhd.ps1
+      arguments: -SubscriptionId $(AZURE_SUBSCRIPTION) `
+                 -Location $(AZURE_LOCATION) `
+                 -ResourceGroupName $(AZURE_RESOURCE_GROUP) `
+                 -ManagedImageName "$(ManagedImageName)" `
+                 -GalleryName "github_imagegeneration_convert_to_vhd" `
+                 -GalleryImageSku "${{ parameters.image_type }}" `
+                 -GalleryImageVersion "0.0.$(Build.BuildId)" `
+                 -StorageAccountName $(AZURE_STORAGE_ACCOUNT) `
+                 -StorageAccountContainerName "images" `
+                 -VhdName "$(VhdName)" `
+                 -ClientId $(CLIENT_ID) `
+                 -ClientSecret $(CLIENT_SECRET) `
+                 -TenantId $(AZURE_TENANT)
+
+  - task: PowerShell@2
+    displayName: 'Remove managed image'
+    inputs:
+      targetType: 'inline'
+      script: |
+        az login --service-principal `
```

Comment:
Can we move this task above "Convert managed image to VHD" and connect to Azure only once?

Reply:
It's not essential since I moved image removal to conversion script.